packages/api/cms-api/src/blocks/video-base.block.ts

```diff
@@ -0,0 +1,51 @@
+import {
+    BlockData,
+    BlockDataInterface,
+    BlockField,
+    BlockInput,
+    ChildBlock,
+    ChildBlockInput,
+    ExtractBlockInput,
+    inputToData,
+} from "@comet/blocks-api";
+import { IsBoolean, IsOptional } from "class-validator";
+
+import { PixelImageBlock } from "../dam/blocks/pixel-image.block";
+
+export class VideoBaseBlockData extends BlockData {
+    @BlockField({ nullable: true })
+    autoplay?: boolean;
+
+    @BlockField({ nullable: true })
+    showControls?: boolean;
+
+    @BlockField({ nullable: true })
+    loop?: boolean;
+
+    @ChildBlock(PixelImageBlock)
+    previewImage: BlockDataInterface;
```

Comment:
Every block that uses this base class needs a migration that adds an empty preview image.